app/apollo/apollo-octopus-public/build.gradle.kts

```diff
@@ -35,8 +35,10 @@ apollo {
     generateOptionalOperationVariables.set(false)
 
     mapScalar("Date", "kotlinx.datetime.LocalDate", "com.apollographql.apollo3.adapter.KotlinxLocalDateAdapter")
+    mapScalar("Instant", "kotlinx.datetime.Instant", "com.apollographql.apollo3.adapter.KotlinxInstantAdapter")
     mapScalarToKotlinString("UUID")
     mapScalarToKotlinString("Url")
     mapScalarToKotlinString("FlowContext")
+    mapScalarToKotlinString("Upload")
```

Comment:
There's `mapScalarToUpload("Upload")` which we use for Giraffe too.
This lets us use Upload https://github.com/apollographql/apollo-kotlin/blob/b4fb9fb7dea17b443fdce0fddce389bfd2331796/libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Upload.kt#L19 from apollo library directly, and it just works regarding setting the content type, length and everything that needs to be done for it to be sent over the network.
Also for us, we get the convenience extension function from `java.io.File.toUpload()` here https://github.com/apollographql/apollo-kotlin/blob/b4fb9fb7dea17b443fdce0fddce389bfd2331796/libraries/apollo-api/src/jvmMain/kotlin/com/apollographql/apollo3/api/FileUpload.kt#L9-L18

Reply:
Excellent